src/audio/host-zephyr.c

```diff
@@ -454,9 +455,20 @@ static int host_copy_normal(struct host_data *hd, struct comp_dev *dev)
 	};
 	notifier_event(hd->chan, NOTIFIER_ID_DMA_COPY,
 		       NOTIFIER_TARGET_CORE_LOCAL, &next, sizeof(next));
-	ret = dma_reload(hd->chan->dma->z_dev, hd->chan->index, 0, 0, copy_bytes);
-	if (ret < 0)
-		comp_err(dev, "host_copy_normal(): dma_copy() failed, ret = %u", ret);
+
+	hd->partial_size += copy_bytes;
+	buffer_c = buffer_acquire(hd->dma_buffer);
+
+	/* Reload DMA when half of the buffer has been used */
```

Comment:
50% because I had no idea and wanted to get some feedback - thank you! :-) Then why 20%? Should it maybe be a minimum of 4 periods and the whole buffer?

Reply:
@mwasko @mmaka1 can you chime in on how this is supposed to work?

Reply:
@lyakh , @yongzhi1 graciously agreed to do a quick experiment with deep buffer playback with this PR and reported a good increase in the lower pc-states residencies.

Reply:
@mmaka1 If we do it in the DMA driver, the threshold should be configurable. Configuration parameters are passed in a `struct dma_config` structure. It doesn't have anything like a "threshold" but it has source and destination burst lengths. Should we use that?

Reply:
@mmaka1 @lyakh What if we have this as a platform setting for now (maybe as a Kconfig)? We could use perhaps DMA attributes, but I wonder if the DMA driver can provide this info. The DMA driver will will likely not know much about how it's integrated to the system (i.e. what's the max latency to access host memory) .

Reply:
@lyakh this looks good to me!